runtime/compiler/runtime/DataCache.cpp

```diff
@@ -249,9 +257,23 @@ TR_DataCache* TR_DataCacheManager::allocateNewDataCache(uint32_t minimumSize)
 
             //--- allocate the segment for the dataCache
             J9MemorySegment *dataCacheSeg = NULL;
+            UDATA memoryType = MEMORY_TYPE_RAM;
+#ifdef LINUX
+            if (_disclaimEnabled)
+               {
+               memoryType |= MEMORY_TYPE_VIRTUAL; // Use mmap for allocation
+               // If swap is enabled, we can allocate memory with mmap(MAP_ANOYNMOUS|MAP_PRIVATE) and disclaim to swap
+               // If swap is not enabled we can disclaim to a backing file
+               TR::CompilationInfo * compInfo = TR::CompilationInfo::get(_jitConfig);
+               if (!TR::Options::getCmdLineOptions()->getOption(TR_DisclaimMemoryOnSwap) || compInfo->isSwapMemoryDisabled())
```

Comment:
What happens if in the checkpoint run swap was enabled, but on restore it is disabled? Does the disclaim attempt on restore just become a nop?

Reply:
Yes, disclaiming a mmaped region with no backing file is  a nop.
The absence of swap is detected upon restore and the disclaiming code will not even attempt to do the disclaim operation.
```
   if (segment->vmemIdentifier.allocator == OMRPORT_VMEM_RESERVE_USED_MMAP_SHM || // Can disclaim to file
       ((segment->vmemIdentifier.mode & J9PORT_VMEM_MEMORY_MODE_VIRTUAL) && canDisclaimOnSwap)) // Can disclaim to swap
 ```